### ISSUE TITLE
Provide dedicated Android support in some form

### ISSUE THREAD

**Issue by Jeffset** [yandex/yatagan#29]
Dagger2 has dagger-android and Hilt, which offer dedicated Android support and simplify injecting dependencies into platform classes.

It's sane to expect Yatagan to provide dedicated Android support in some way. It probably will not be directly compatible with either dagger-android or Hilt, because both are implemented as a separate annotation-processors, which introduces multiple rounds of processing and penalize the build speed. Moreover, Hilt is an aggregating AP so implementing something like that with runtime reflection is impossible.

If Yatagan adds specific support for Android, it would be more close to dagger-android approach, then to Hilt's, though anything can change.

TL; DR; Yatagan has plans to support Android and discussions about the design and the API are going to be happening here.
